src/NServiceBus.Core/Transports/TransportReceiver.cs

```diff
@@ -61,19 +64,42 @@ class TransportReceiver
 
         async Task InvokePipeline(PushContext pushContext)
         {
+            //note: we should get this value from the transport in the future to make it more accurate
+            var pipelineStartedAt = DateTime.UtcNow;
+
+
             using (var childBuilder = builder.CreateChildBuilder())
             {
                 var rootContext = new RootContext(childBuilder, pipelineCache, eventAggregator);
-                var context = new TransportReceiveContext(new IncomingMessage(pushContext.MessageId, pushContext.Headers, pushContext.BodyStream), pushContext.TransportTransaction, pushContext.ReceiveCancellationTokenSource, rootContext);
+
+                //notifications are only relevant for the main pipeline since satellites pipelines are not exposed to user in any way
+                if (isMainReceiver)
+                {
+                    var sw = new Stopwatch();
+
+                    sw.Start();
+
+                    // float the sw via the context to avoid threading issues should the continuation where we access 
+                    // it is executed on a different thread
+                    rootContext.Set(sw);
```

Comment:
To be clear, it's not the `StopWatch` settings but a system setting using [QueryPerformanceFrequency](https://msdn.microsoft.com/en-us/library/windows/desktop/ms644905%28v=vs.85%29.aspx). If the hardware enables high frequency counter, then `StopWatch` will use it. Otherwise it will fallback to `DateTime.UtcNow.Ticks`. It's highly unlikely that it will fallback to ticks now on modern machines. This will result in using the high frequency performance counter which AFAIK can suffer from stale reads if used in multiple thread.
The piece of code we're discussing has one await before obtaining `sw.Elapsed` hence it may be run on a different thread hence it might suffer from the clock skew.